language/extensions/move-table-extension/sources/Table.move

```diff
@@ -43,12 +43,22 @@ module Extensions::Table {
         &borrow_box<K, V, Box<V>>(table, key).val
     }
 
+    /// Same as `borrow` but pass in a copyable key instead.
+    public fun borrow_by_key<K: copy, V>(table: &Table<K, V>, key: K): &V {
+        &borrow_box_by_key<K, V, Box<V>>(table, key).val
```

Comment:
doesn't this work `&native_borrow_box<K, V, Box<V>>(table, &key).val`?

Do you need to introduce `borrow_box_by_key`?

Reply:
K. So native functions are checked as well. You won. 😂 
Do some code reuse on the native side at least, though.

Reply:
It doesn't what because it surfaces the same problem we try to solve here.